src/python/pants/option/options_test.py

```diff
@@ -353,8 +353,9 @@ def test_deprecated_options_error() -> None:
     def register(opts: Options) -> None:
         opts.register(GLOBAL_SCOPE, "--expired", removal_version="0.0.1.dev0")
 
+    opts = create_options([GLOBAL_SCOPE], register, [])
     with pytest.raises(CodeRemovedError):
-        create_options([GLOBAL_SCOPE], register, [])
+        opts.for_scope(scope=GLOBAL_SCOPE)
```

Comment:
creating options doesn't now trigger the expired deprecation check, we must parse.